### PR TITLE
feature: ZENKO-1438 Add conditional MD put tests

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Depends on https://github.com/scality/Arsenal/pull/655.

### CODE REVIEW

File: tests/functional/metadata/MongoClientInterface.js [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+            } else {
+                assert(result.tag === TAG_1);
+                assert.deepStrictEqual(result.value, objVal);
+            }
+            return cb();
+        });
+    }
+
+    beforeEach(done =>
+        MongoClient.connect(mongoUrl, {}, (err, client) => {
```
Comment (jonathan-gramain) [inline tests/functional/metadata/MongoClientInterface.js:90]: minor: do we have to reconnect / recreate a new db on each test? Using different object keys with `uuid` should be enough I think.

File: tests/functional/metadata/MongoClientInterface.js [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        it('should update metadata when no tag is provided', done =>
+            testPutMetadata({
+                params: {},
+                shouldHaveUpdated: true,
+            }, done));
+
+        it('should update metadata when matching tag is provided', done =>
+            testPutMetadata({
+                params: {
+                    condPut: {
```
Comment (jonathan-gramain) [inline tests/functional/metadata/MongoClientInterface.js]: heads-up to rename this field to `cond`

File: tests/functional/metadata/MongoClientInterface.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        const db = mongoClient.db(TEST_DB);
+        return db.dropDatabase(err => {
+            if (err) {
+                return done(err);
+            }
+            return mongoClient.close(true, done);
+        });
+    });
+
+    describe('::putObjectNoVer', () => {
```
Comment (jonathan-gramain) [inline tests/functional/metadata/MongoClientInterface.js:122]: Do these functions `putObjectNoVer` etc. allow creation of new metadata and return a new tag? If so, could be worth adding unit tests for new insertions.

Comment (bennettbuchanan) [reply tests/functional/metadata/MongoClientInterface.js:122]: Updated.

**Comment by bennettbuchanan** [comment]
Putting this on hold because https://scality.atlassian.net/browse/ZENKOIO-11 is taking precedence in the coming weeks. We can reopen should there be an immediate need for these changes.

**Pull request closed without merging**